alicloud/service_alicloud_polardb.go

```diff
@@ -0,0 +1,801 @@
+	parts, err := ParseResourceId(id, 3)
+	if err != nil {
+		return nil, WrapError(err)
+	}
+	deadline := time.Now().Add(time.Duration(DefaultIntervalLong) * time.Second)
+	for {
+		object, err := s.DescribePolarDBInstanceNetInfo(parts[0])
+
+		if err != nil {
+			if NotFoundError(err) {
+				return nil, WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
+			}
+			return nil, WrapError(err)
+		}
+
+		if object != nil {
+			for _, o := range object {
+				if o.DBEndpointId == parts[1] {
+					for _, p := range o.AddressItems {
+						if p.NetType == parts[2] {
+							return &p, nil
+						}
+					}
+				}
+			}
+		}
+		time.Sleep(DefaultIntervalMini * time.Second)
+		if time.Now().After(deadline) {
+			break
+		}
+	}
+
+	return nil, WrapErrorf(Error(GetNotFoundMessage("DBConnection", id)), NotFoundMsg, ProviderERROR)
+}
+
+func (s *PolarDBService) DescribePolarDBInstanceNetInfo(id string) ([]polardb.DBEndpoint, error) {
+
+	request := polardb.CreateDescribeDBClusterEndpointsRequest()
+	request.RegionId = s.client.RegionId
+	request.DBClusterId = id
+	raw, err := s.client.WithPolarDBClient(func(polarDBClient *polardb.Client) (interface{}, error) {
+		return polarDBClient.DescribeDBClusterEndpoints(request)
+	})
+
+	if err != nil {
+		if IsExceptedErrors(err, []string{InvalidDBClusterIdNotFound, InvalidDBClusterNameNotFound}) {
+			return nil, WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
+		}
+		return nil, WrapErrorf(err, DefaultErrorMsg, id, request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+
+	response, _ := raw.(*polardb.DescribeDBClusterEndpointsResponse)
+	if len(response.Items) < 1 {
+		return nil, WrapErrorf(Error(GetNotFoundMessage("DBInstanceNetInfo", id)), NotFoundMsg, ProviderERROR)
+	}
+
+	return response.Items, nil
+}
+
+func (s *PolarDBService) DescribePolarDBDatabase(id string) (ds *polardb.Database, err error) {
+	parts, err := ParseResourceId(id, 2)
+	if err != nil {
+		return nil, WrapError(err)
+	}
+	dbName := parts[1]
+	request := polardb.CreateDescribeDatabasesRequest()
+	request.RegionId = s.client.RegionId
+	request.DBClusterId = parts[0]
+	request.DBName = dbName
+
+	var raw interface{}
+	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
+		raw, err = s.client.WithPolarDBClient(func(polarDBClient *polardb.Client) (interface{}, error) {
+			return polarDBClient.DescribeDatabases(request)
+		})
+		if err != nil {
+			if IsExceptedErrors(err, []string{DBInternalError, InvalidDBClusterStatus, InvalidDBClusterIdNotFound}) {
+				time.Sleep(10 * time.Second)
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+		return nil
+	})
+
+	if err != nil {
+		if IsExceptedErrors(err, []string{DBInternalError, InvalidDBClusterStatus, InvalidDBClusterIdNotFound}) {
```

Comment:
I think the `DBInternalError ` and `InvalidDBClusterStatus` do not belong to `NotFoundError`